src/libxtp/aomatrices/aocoulomb.cc

```diff
@@ -1143,52 +1143,39 @@ if (_lmax_col > 5) {
     
 
     
-    int AOCoulomb::Symmetrize(const AOOverlap& _gwoverlap,const AOBasis& gwbasis, ub::matrix<double>& _gwoverlap_inverse, ub::matrix<double>& _gwoverlap_cholesky_inverse){
-        int removed_functions=0;
-        if ( gwbasis._is_stable ){
-            
-            // get inverse of _aooverlap
-            
-           removed_functions=linalg_invert_svd( _gwoverlap.Matrix(), _gwoverlap_inverse,1e7 );
-            
-            
-            // make copy of _gwoverlap, because matrix is overwritten in GSL
-            ub::matrix<double> _gwoverlap_cholesky = _gwoverlap.Matrix();
-            linalg_cholesky_decompose( _gwoverlap_cholesky );
-           
-            // remove L^T from Cholesky
-            #pragma omp parallel for 
-            for (unsigned i =0; i < _gwoverlap_cholesky.size1(); i++ ){
-                for (unsigned j = i+1; j < _gwoverlap_cholesky.size1(); j++ ){
-                    _gwoverlap_cholesky(i,j) = 0.0;
-                }
-            }
-            
-            // invert L to get L^-1
-           
-            int removed2=linalg_invert_svd(  _gwoverlap_cholesky, _gwoverlap_cholesky_inverse,1e7 );
-            if(removed2>removed_functions){
-                removed_functions=removed2;
-            }
-            
-            // calculate V' = L^-1 V (L^-1)^T
-            ub::matrix<double> _temp = ub::prod( _gwoverlap_cholesky_inverse , _aomatrix );
-            _aomatrix = ub::prod( _temp, ub::trans(_gwoverlap_cholesky_inverse));
-            
-            linalg_matrixsqrt(_aomatrix);
-           
-            // multiply with L from the left and L+ from the right
-            _temp = ub::prod( _gwoverlap_cholesky , _aomatrix );
-            _aomatrix = ub::prod( _temp ,ub::trans( _gwoverlap_cholesky ));
-            
-            _aomatrix = ub::prod(_aomatrix , _gwoverlap_inverse);
- 
-
-        }
-        else{
-            cout<<"WARNING: gwbasis is not stable."<<endl;
+    int AOCoulomb::Symmetrize(const ub::matrix<double>& _gwoverlap_cholesky){
+        
+       //This converts V into L(LT V L)-1/2 LT, which is needed to construct 4c integrals,
```

Comment:
Are you documenting this somewhere in more detail? We need that for eventual alternatives to PPM.

Reply:
To be honest, I have no real clue, why this worked. The non orthogonal space has two representations connected by the S matrix. I think without the ppm you do not have to use this expression but V-1/2 might be enough.